openmc/geometry.py

```diff
@@ -389,7 +389,7 @@ def get_all_universes(self) -> typing.Dict[int, openmc.Universe]:
         """
         universes = {}
         universes[self.root_universe.id] = self.root_universe
-        universes.update(self.root_universe.get_all_universes())
+        universes.update(self.root_universe.get_all_universes(memo=set()))
```

Comment:
With the memoization pattern, there shouldn't be a need to create the memo argument at the top level. Creating an empty set (or dictionary?) should be done in the first function that discovers that the `memo` argument passed in was `None`.

Reply:
Hmm I don't disagree, but I do see this pattern elsewhere in `get_all_cells`, `get_all_materials`, etc. Would you like me to update those methods on the `Cell`, `Universe`, and `Lattice` objects here?

Reply:
On it!